core/ecschema-metadata/src/UnitQuery/UnitQuery.ts

```diff
@@ -0,0 +1,171 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { BentleyError, BentleyStatus } from "@bentley/bentleyjs-core";
+import { SchemaContext, SchemaItem, SchemaItemKey, SchemaKey, Unit } from "../ecschema-metadata";
+import { SchemaItemType } from "../ECObjects";
+import { UNIT_DATA } from "./UnitsData";
```

Comment:
we would want this data to come from outside of the ecschema-metadata package.  So it needs to get passed in as data to any functions we create in ecschema-metadata

Reply:
Changed UnitQuery class to take in extra data associated with Units in constructor so that data may be passed in wherever UnitQuery is used.